### DISCUSSION TITLE
node values updating within selectmany

### DISCUSSION THREAD

**Discussion by jbscherrer** [bonsai-rx/bonsai#1367] [Q&A] [answered]
I'm trying to create a SelectMany node that takes in sets of doubles and writes them to individual .csv files, with the names of the files determined by the time at which each double arrives. I thought an InputMapping would be the correct way to do this, as my understanding is that the InputMapping ensures all properties are updated before the node processes the corresponding input. However, the below workflow does not behave as expected, and instead each double is saved in a file with the timestamp of the previous double, suggesting that the CsvWriter node is processing the input without first updating the FileName property. Is my understanding of InputMapping incorrect? If so, what is the appropriate way to do this?

![image](https://github.com/bonsai-rx/bonsai/assets/39060975/e644fe59-f38c-44ac-bfc2-7efc89ebe986)

**Comment by bruno-f-cruz** [comment, marked as answer]
![image](https://github.com/bonsai-rx/bonsai/assets/7049351/0acce711-8841-4cc7-87a4-f73561268a89)
tldr: Re-use Subjects and avoid branching :P. [Check this one for a lengthier explanation](https://groups.google.com/g/bonsai-users/c/lEj2W_5qgfQ/m/M_hXa_OkAAAJ).

**Reply by jbscherrer** [reply]
Ah, thank you! I think this graphic or similar would be a helpful addition to the [Workflow Design Guidelines](https://bonsai-rx.org/docs/articles/workflow-guidelines.html).
